framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenes.java

```diff
@@ -192,6 +210,112 @@ private void updateInferredExecutableParameterTypes(
         }
     }
 
+    @Override
+    public void updateContracts(
+            Analysis.BeforeOrAfter preOrPost,
+            ExecutableElement methodElt,
+            CFAbstractStore<?, ?> store) {
+
+        if (store == null) {
+            throw new BugInCF(
+                    "updateContracts(%s, %s, null) for %s",
+                    preOrPost, methodElt, atypeFactory.getClass().getSimpleName());
+        }
+
+        String className = getEnclosingClassName(methodElt);
+        String jaifPath = storage.getJaifPath(className);
+        AClass clazz =
+                storage.getAClass(className, jaifPath, ((MethodSymbol) methodElt).enclClass());
+        AMethod amethod = clazz.methods.getVivify(JVMNames.getJVMMethodSignature(methodElt));
+
+        amethod.setFieldsFromMethodElement(methodElt);
+
+        // TODO: Probably move some part of this into the AnnotatedTypeFactory.
+
+        // TODO: Handle more than just fields of "this".
+
+        // Process the store's fields.
+        for (Map.Entry<FieldAccess, ? extends CFAbstractValue<?>> entry :
+                store.getFieldValues().entrySet()) {
+            FieldAccess fa = entry.getKey();
+            CFAbstractValue<?> v = entry.getValue();
+
+            VariableElement fieldElement = fa.getField();
+            AnnotatedTypeMirror fieldType = atypeFactory.getAnnotatedType(fieldElement);
+
+            if (!ElementUtils.inSameClass(methodElt, fieldElement)) {
+                continue;
+            }
+
+            AnnotatedTypeMirror atm = convertCFAbstractValueToAnnotatedTypeMirror(v, fieldType);
+            adjustForUpdateNonField(atm);
+
+            AField afield = vivifyAndAddTypeMirrorToContract(amethod, preOrPost, fieldElement);
+
+            updateAnnotationSetInScene(
+                    afield.type, TypeUseLocation.FIELD, atm, fieldType, jaifPath, false);
+        }
+
+        // Process fields that are not in the store.
+        TypeElement containingClass = (TypeElement) methodElt.getEnclosingElement();
+        ThisReference thisReference = new ThisReference(containingClass.asType());
+        ClassName classNameReceiver = new ClassName(containingClass.asType());
+        for (VariableElement fieldElement :
+                ElementFilter.fieldsIn(containingClass.getEnclosedElements())) {
+            FieldAccess fa =
+                    new FieldAccess(
+                            (ElementUtils.isStatic(fieldElement)
+                                    ? classNameReceiver
+                                    : thisReference),
+                            fieldElement.asType(),
+                            fieldElement);
+            if (store.getFieldValue(fa) == null) {
```

Comment:
Instead of looping through all the field accesses in the store above, could you just add an else branch here that does the same thing as the code above?